packages/server/lib/errors.coffee

```diff
@@ -575,7 +575,15 @@ getMsgByType = (type, arg1 = {}, arg2, arg3) ->
       return msg
     when "PLUGINS_DIDNT_EXPORT_FUNCTION"
       msg = """
-      The `pluginsFile` must export a function.
+      The `pluginsFile` must export a function with the following signature:
+
+      ```
+      module.exports = function (on, config) {
+        // configure plugins here
+      }
+      ```
+
+      Learn more: https://on.cypress.io/writing-a-plugin
```

Comment:
For on links, we create a specific on link for each error message. Since we cannot ever update previous versions of Cypress - these links will be static. So if this 'writing-a-plugin' guide ever changes its filename, this link in older versions would be dead. 

The exception to this rule is mostly when linking directly to api docs (like `https://on.cypress.io/click`) cause these aren't as subjective as where guides are placed.

This should probably be explained in our contributing doc somewhere. 😬 I'll add a section.

Anyways, 

- You just make up some reasonable name for the link `https://on.cypress.io/plugins-must-export-function`. 
- Add your new slug to this file in services https://github.com/cypress-io/cypress-services/blob/develop/packages/on/data/links.yml having the link redirect to `writing-a-plugin` url and open PR for their repo.
- They will have to accept the PR AND **deploy** for the link redirect to work. 
- Then this issue can be merged.

Reply:
I am awaiting an access request for `cypress-services`.

Reply:
I've submitted a PR here: https://github.com/cypress-io/cypress-services/pull/2590